crates/project_panel/src/project_panel.rs

```diff
@@ -433,6 +441,32 @@ impl ProjectPanel {
         })
     }
 
+    fn update_diagnostics(&mut self, cx: &mut ViewContext<Self>) {
+        self.diagnostic_error_colors = self
+            .project
+            .read(cx)
+            .diagnostic_summaries(false, cx)
+            .filter_map(|(path, _, diagnostic_summary)| {
+                if diagnostic_summary.error_count > 0 {
+                    Some((path, Color::Error))
+                } else {
+                    None
+                }
```

Comment:
I thought about doing this:
```rust
if diagnostic_summary.error_count > 0 {
    Some((path, Color::Error))
} else if else if diagnostic_summary.warning_count > 0 {
    Some((path, Color::Warning))
} else {
    None
}
```
I didn't like that and removed it. We could make this configurable if you want?